vllm/v1/core/sched/scheduler.py

```diff
@@ -948,7 +985,8 @@ def _free_encoder_inputs(self, request: Request) -> None:
             mm_positions = request.mm_positions[input_id]
             start_pos = mm_positions.offset
             num_tokens = mm_positions.length
-            if start_pos + num_tokens <= request.num_computed_tokens:
+            if (self.encoder_cache_manager
```

Comment:
Is it necessary to test `self.encoder_cache_manager` here again since it's already tested at the top of the function?

Reply:
good catch!